src/test/scala/chiselTests/AutoClonetypeSpec.scala

```diff
@@ -172,4 +172,15 @@ class AutoClonetypeSpec extends ChiselFlatSpec {
       val a = WireDefault(io)
     } }
   }
+
+  "Aliased fields" should "be caught" in {
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val bundleFieldType = UInt(8.W)
+        val io = IO(Output(new Bundle {
+          val a = bundleFieldType
+        }))
+      } }
```

Comment:
Is the following also subject to issues related to aliased fields?

```scala
new Module {
        val io = IO(Output(new Bundle {
          val a = UInt(8.W)
        }))
      }
```

Reply:
Shouldn't, because it creates a new UInt on the constructor invocation, and auto-clone-type constructs a new Bundle each time.

Reply:
So the following would also have aliasing issues?
```scala
class AliasedBundle(foo: UInt) extends Bundle {
  val a = foo
  val b = foo
}
```

Reply:
Yes, but I don't think that's in the domain of auto-cloneType to check, and under current code it will probably cause a rebinding error anyways.

Reply:
What I don't understand is why autoclonetype can't check for `AliasedBundle` when it can check for the example in your test...?

Reply:
It's not a automatically generated clone type issue (as in, issue between the original bundle and cloned bundle), it's your Bundle definition is wrong (problem is solely within one Bundle object). And I think it's also caught by the rebinding checks. If you want I'll add a test case for that?

Reply:
Sounds good